common/artifact-manager/src/main/java/com/vmware/pscoe/iac/artifact/store/vrang/VraNgPropertyGroupStore.java

```diff
@@ -60,68 +72,83 @@ public void init(RestClientVraNg restClient, Package vraNgPackage, Configuration
+		if (!propertyGroupFolder.exists()) {
+			logger.info("Property Group Dir not found.");
 			return;
 		}
 
-		File[] propertyGroupFiles= this.filterBasedOnConfiguration(propertyGroupFolder, new CustomFolderFileFilter(this.getItemListFromDescriptor()));
-		if ( propertyGroupFiles == null || propertyGroupFiles.length == 0) {
-			logger.info( "Could not find any property groups." );
+		File[] propertyGroupFiles = this.filterBasedOnConfiguration(propertyGroupFolder,
+				new CustomFolderFileFilter(this.getItemListFromDescriptor()));
+		if (propertyGroupFiles == null || propertyGroupFiles.length == 0) {
+			logger.info("Could not find any property groups.");
 			return;
 		}
 
-		logger.info( "Found property groups. Importing..." );
+		logger.info("Found property groups. Importing...");
 
 		for (File file : propertyGroupFiles) {
-			logger.info( file.getName() );
-			logger.info( file.getAbsolutePath() );
-			VraNgPropertyGroup propertyGroup	= jsonFileToVraPropertyGroup( file );
-			importPropertyGroup( propertyGroup, propertyGroupFolder );
+			logger.info(file.getName());
+			logger.info(file.getAbsolutePath());
+			VraNgPropertyGroup propertyGroup = jsonFileToVraPropertyGroup(file);
+			importPropertyGroup(propertyGroup, propertyGroupFolder);
 		}
 	}
 
 	/**
 	 * Imports a specific property group.
 	 * First checks if a property group with given name already exists,
-	 * so it gets its id and based on that performs either create or update operation.
+	 * so it gets its id and based on that performs either create or update
+	 * operation.
 	 * If a property group doesn't exist set project ID in request.
-	 * If a property group exists, set the property group id in the request, so vRA knows which PG to update
+	 * If a property group exists, set the property group id in the request, so vRA
+	 * knows which PG to update
 	 *
 	 * @param propertyGroup
 	 * @param propertyGroupsFolder
 	 */
 	private void importPropertyGroup(VraNgPropertyGroup propertyGroup, File propertyGroupsFolder) {
 		File customPropertyGroupFile = Paths.get(propertyGroupsFolder.getPath(), getName(propertyGroup)).toFile();
+		String organizationId = VraNgOrganizationUtil.getOrganization(this.restClient, this.config).getId();
+
 		VraNgPropertyGroup existingPropertyGroup = this.getPropertyGroupByName(propertyGroup.getName());
 
-		if (existingPropertyGroup != null) {
-			if (!existingPropertyGroup.hasTheSameProjectId(this.projectId)) {
-				logger.warn("The property group {} is already assigned to a different project. Property group scope will NOT be changed and the group will NOT be updated.", getName(propertyGroup));
-			} else {
-				propertyGroup.setId(existingPropertyGroup.getId());
-				logger.info("Updating property group: {}", customPropertyGroupFile.getAbsolutePath());
-				this.restClient.updatePropertyGroup(propertyGroup);
+		if (!propertyGroup.hasTheSameOrgId(organizationId)) {
```

Comment:
orgId should be taken from the configuration (settings.xml and or installer properties) and should always be set.

Essentially if the property group that is stored on the machine has orgId set in the JSON, it needs to always be set to the one coming from the configuration

Reply:
Yes, if it is different, we override it always. Nothing has to be changed.
If it is the same (export is done from the same system) we do not override, because there is no need.
We need the organisation id, to be able to identify the case, where the organisation is the same, but the project is different, and in this case we generate an error message and do not update the property group.